source/Music.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <thread>
+#include "Music.hpp"
+#include "Audio.hpp"
+#include "AudioError.hpp"
+
+Music::Music(Musics m, float loop)
+  : source(AL_NONE), loopTime(loop)
+{
+  if (ov_fopen(Audio::getMusicFileName(m), &oggStream) < 0)
+    throw AudioError("Could not open .ogg file");
+  vorbisInfo = ov_info(&oggStream, -1);
+  vorbisComment = ov_comment(&oggStream, -1);
+  format = vorbisInfo->channels == 1 ? AL_FORMAT_MONO16 : AL_FORMAT_STEREO16;
+  alGenSources(1, &source);
+  Audio::checkError();
+  alGenBuffers(2, buffers.data());
+  if (!Audio::checkError(false))
+    {
+      alDeleteSources(1, &source);
+      ov_clear(&oggStream);
+      throw AudioError("Couldn't generate 1 buffer in Music::Music");
+    }
+
+  alSource3f(source, AL_POSITION, 0., 0., 0.);
+  alSource3f(source, AL_VELOCITY, 0., 0., 0.);
+  alSource3f(source, AL_DIRECTION, 0., 0., 0.);
+  alSourcef(source, AL_ROLLOFF_FACTOR, 0.);
+  alSourcef(source, AL_SOURCE_RELATIVE, AL_TRUE);
+  if (!Audio::checkError(false))
+    {
+      alDeleteSources(1, &source);
+      ov_clear(&oggStream);
+      alDeleteBuffers(1, &buffers[0]);
+      throw AudioError("Couldn't generate 1 buffer in Music::Music");
+    }
+}
+
+Music::~Music()
+{
+  alSourceStop(source);
+  unqueuePending();
+  alDeleteSources(1, &source);
+  alDeleteBuffers(1, &buffers[0]);
+  ov_clear(&oggStream);
+}
+
+bool Music::play(void)
+{
+  if (!streamFile(buffers[0]) || !streamFile(buffers[1]))
+    return false;
+  alSourceQueueBuffers(source, 2, buffers.data());
+  alSourcePlay(source);
+  return true;
+}
+
+void Music::setLoopTime(float f)
+{
+  loopTime = f;
+}
+
+bool Music::isPlaying(void) const
+{
+  ALenum state;
+
+  alGetSourcei(source, AL_SOURCE_STATE, &state);
+  return state == AL_PLAYING;
+}
+void Music::update(void)
+{
+  ALuint buffer = AL_NONE;
+  int processed = AL_NONE;
+  bool active = true;
+
+  alGetSourcei(source, AL_BUFFERS_PROCESSED, &processed);
+  while (processed--)
+    {
+      alSourceUnqueueBuffers(source, 1, &buffer);
+      Audio::checkError();
+      active = streamFile(buffer);
```

Comment:
This seems weird. `active` will be set at each iteration of the while, independent of last one.
Is this really what is intended?